dev/WebView2/WebView2.cpp

```diff
@@ -938,115 +905,88 @@ void WebView2::FillPointerInfo(const winrt::PointerPoint& inputPt, winrt::CoreWe
+    outputPt.HistoryCount(args.GetIntermediatePoints(*this).Size());
+
+    LARGE_INTEGER lpFrequency{};
+    if (QueryPerformanceFrequency(&lpFrequency))
     {
-        outputPt_pointerFlags |= POINTER_FLAG_INRANGE;
+        auto scale = 1000000;
+        auto frequency = lpFrequency.QuadPart;
+        outputPt.PerformanceCount((inputPt.Timestamp() * frequency) / scale);
     }
 
+    outputPt.ButtonChangeKind(static_cast<int32_t>(inputProperties.PointerUpdateKind()));
+}
+
+uint32_t WebView2::GetPointerFlags(const winrt::PointerPoint& inputPt)
+{
+    winrt::PointerPointProperties inputProperties{ inputPt.Properties() };
+    winrt::PointerDeviceType deviceType{ inputPt.PointerDevice().PointerDeviceType() };
+    uint32_t pointerFlags{ POINTER_FLAG_NONE };
+
     if (deviceType == winrt::PointerDeviceType::Touch)
     {
         if (inputPt.IsInContact())
         {
-            outputPt_pointerFlags |= POINTER_FLAG_INCONTACT;
-            outputPt_pointerFlags |= POINTER_FLAG_FIRSTBUTTON;
+            pointerFlags |= POINTER_FLAG_INCONTACT;
+            pointerFlags |= POINTER_FLAG_FIRSTBUTTON;
         }
 
         if (inputProperties.PointerUpdateKind() == winrt::PointerUpdateKind::LeftButtonPressed)
         {
-            outputPt_pointerFlags |= POINTER_FLAG_NEW;
+            pointerFlags |= POINTER_FLAG_NEW;
         }
     }
-
-    if (deviceType == winrt::PointerDeviceType::Pen)
+    else if (deviceType == winrt::PointerDeviceType::Pen)
     {
         if (inputPt.IsInContact())
         {
-            outputPt_pointerFlags |= POINTER_FLAG_INCONTACT;
+            pointerFlags |= POINTER_FLAG_INCONTACT;
 
             if (!inputProperties.IsBarrelButtonPressed())
             {
-                outputPt_pointerFlags |= POINTER_FLAG_FIRSTBUTTON;
+                pointerFlags |= POINTER_FLAG_FIRSTBUTTON;
             }
-
             else
             {
-                outputPt_pointerFlags |= POINTER_FLAG_SECONDBUTTON;
+                pointerFlags |= POINTER_FLAG_SECONDBUTTON;
             }
         } // POINTER_FLAG_NEW is currently omitted for pen input
     }
 
-    if (inputProperties.IsPrimary())
-    {
-        outputPt_pointerFlags |= POINTER_FLAG_PRIMARY;
-    }
+    if (inputProperties.IsInRange()) { pointerFlags |= POINTER_FLAG_INRANGE; }
+    if (inputProperties.IsPrimary()) { pointerFlags |= POINTER_FLAG_PRIMARY; }
+    if (inputProperties.IsCanceled()) { pointerFlags |= POINTER_FLAG_CANCELED; }
+    if (inputProperties.TouchConfidence()) { pointerFlags |= POINTER_FLAG_CONFIDENCE; }
+    if (inputProperties.PointerUpdateKind() == winrt::PointerUpdateKind::LeftButtonPressed) { pointerFlags |= POINTER_FLAG_DOWN; }
+    if (inputProperties.PointerUpdateKind() == winrt::PointerUpdateKind::LeftButtonReleased) { pointerFlags |= POINTER_FLAG_UP; }
+    if (inputProperties.PointerUpdateKind() == winrt::PointerUpdateKind::Other) { pointerFlags |= POINTER_FLAG_UPDATE; }
```

Comment:
There is a change for WinUI2 where we are reverting the usage of PointerUpdateKind() for processing mouse as it is deemed unreliable. Is that problem also applicable here?

Reply:
This is only a whitespace change. The if statement is the same, just all on one line for readability. If there's a bug here (and I don't think there is, that was mouse, this is only for pen and touch) let's fix it with a different change.

Reply:
Agree this would be an existing issue - not blocking this change. Separately, it'd be good to confirm the PointerUpdateKind() sceanrio that was problematic with mouse does not repro for touch/pen as used here. I'll follow up offline.

Reply:
http://task.ms/40916025